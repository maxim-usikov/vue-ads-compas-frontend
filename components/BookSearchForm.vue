<template>
  <VForm ref="form" v-model="valid" @submit.prevent="handleSearch">
    <VTextField
      v-model="q"
      label="Book title or author name"
      :rules="qRules"
      required
    />

    <!-- this preformated by eslint/prettier ... -->
    <VBtn color="primary" :disabled="!valid || loading" @click="handleSearch"
      >Search</VBtn
    >
  </VForm>
</template>

<script>
export default {
  name: 'BookSearchForm',

  props: {
    loading: {
      type: Boolean,
      default: false
    }
  },

  data() {
    return {
      valid: false,
      q: this.$route.query.q || '',
      qRules: [
        v => !!v || 'Required',
        v => v.length > 2 || 'Must be greater then 2 characters'
      ]
    }
  },

  methods: {
    handleSearch() {
      const { q } = this

      this.$emit('search', { q })
    }
  }
}
</script>
