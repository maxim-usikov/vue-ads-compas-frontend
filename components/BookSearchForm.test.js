import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'

import BookSearchForm from './BookSearchForm.vue'

const localVue = createLocalVue()
localVue.use(VueRouter)

describe('BookSearchForm', () => {
  let $c

  beforeEach(() => {
    const router = new VueRouter({})
    $c = shallowMount(BookSearchForm, { localVue, router })
  })

  it('is a Vue instance', () => {
    expect($c.isVueInstance()).toBeTruthy()
  })

  it('name is `BookSearchForm`', () => {
    expect($c.name()).toBe('BookSearchForm')
  })

  it('renders correctly', () => {
    expect($c.element).toMatchSnapshot()
  })
})
