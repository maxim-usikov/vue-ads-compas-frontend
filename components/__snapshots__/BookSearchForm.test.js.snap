// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookSearchForm renders correctly 1`] = `
<vform-stub>
  <vtextfield-stub
    label="Book title or author name"
    required=""
    rules="v => {
          /* istanbul ignore next */
          cov_8j5301kt1.f[2]++;
          cov_8j5301kt1.s[5]++;
          return (
            /* istanbul ignore next */
            (cov_8j5301kt1.b[1][0]++, !!v) ||
            /* istanbul ignore next */
            (cov_8j5301kt1.b[1][1]++, 'Required')
          );
        },v => {
          /* istanbul ignore next */
          cov_8j5301kt1.f[3]++;
          cov_8j5301kt1.s[6]++;
          return (
            /* istanbul ignore next */
            (cov_8j5301kt1.b[2][0]++, v.length > 2) ||
            /* istanbul ignore next */
            (cov_8j5301kt1.b[2][1]++, 'Must be greater then 2 characters')
          );
        }"
    value=""
  />
   
  <vbtn-stub
    activeclass="v-btn--active"
    color="primary"
    disabled="true"
    tag="button"
    type="button"
  >
    Search
  </vbtn-stub>
</vform-stub>
`;
